outbox-core/src/main/java/io/praesidio/outbox/Outbox.java

```diff
@@ -1,29 +1,22 @@
 package io.praesidio.outbox;
 
-import io.praesidio.outbox.values.MessageType;
 import lombok.NonNull;
 
 import java.util.Collection;
-import java.util.Map;
-import java.util.Optional;
-import java.util.stream.Collectors;
 
 public class Outbox {
 
+    private final MessageFactory messageFactory;
     private final MessageRepository messageRepository;
-    private final Map<MessageType, MessageSerializer> serializers;
 
     public Outbox(@NonNull MessageRepository messageRepository,
                   @NonNull Collection<MessageSerializer> serializers) {
         this.messageRepository = messageRepository;
-        this.serializers = serializers.stream().collect(Collectors.toMap(MessageSerializer::getType, s -> s));
+        this.messageFactory = new MessageFactory(serializers);
```

Comment:
I am not sure if it is good - but I did it like these becasue I thought that we don't want to make MessageFactory public.